chrome/content/zotero/zoteroPane.js

```diff
@@ -1893,6 +1909,21 @@ var ZoteroPane = new function()
 		}
 		this.selectItem(item.id).done();
 	}
+
+
+	/**
+	 * Return whether every selected item is deleted.
+	 *
+	 * @return {Boolean}
+	 */
+	this.canRestoreSelectedItems = function () {
+		let collectionTreeRow = this.getCollectionTreeRow();
+		if (!collectionTreeRow.isTrash()) {
+			return false;
+		}
+
+		return this.getSelectedItems().every(item => item.deleted);
```

Comment:
This isn't quite right. If you select a parent item and a child item in the trash, "Restore to Library" should be enabled, even if one of them isn't technically deleted.

Reply:
I'd like to get the logic right here, but it depends on what we're going for - should we only show Restore if it will cause everything you have selected to disappear from the trash? If so, we also need to check whether any of the selected items has a descendant that is deleted and not selected, since that would mean that part of your selection will remain in the trash after Restore. If we only care whether it'll clear _some_ of the selected items from the trash, the test can be simpler.

Reply:
- Deleted/selected parent, deleted/selected child, no other children → show Restore, undelete both
- Non-deleted/selected parent, deleted/selected child, no other children → show Restore, undelete child (because there's no reason not to)
- Non-deleted/selected parent, deleted/selected child, deleted/non-selected children → show Restore, undelete selected child, don't undelete non-selected children (because the intention is clear)
- Deleted/selected parent, deleted/non-selected child, non-deleted/non-selected child → show Restore, restore parent and deleted child (because there's no way to know the user's intention and this is safer)
- Deleted/selected parent, non-deleted/selected child, deleted/non-selected child → show Restore, restore parent only, don't undelete non-selected child (because selecting a specific child communicates intention)
- Non-deleted/selected parent, deleted/non-selected child → don't show Restore (because they didn't select anything non-gray)
- Non-deleted/non-selected parent, deleted/selected child, non-deleted/selected child → show Restore, undelete deleted child (because non-deleted items in the selection don't prevent Restore)

Reply:
(Let's create tests for all of these.)

Reply:
See latest commits!